security/boot-security-gorm/src/main/groovy/yakworks/security/gorm/model/AppUserRepo.groovy

```diff
@@ -78,8 +82,35 @@ class AppUserRepo implements GormRepo<AppUser> {
         if(user.password) {
             user.passwordHash = encodePassword(user.password)
         }
+        if(user.isNew()) {
+            //we check when new to avoid unique index error.
+            if(exists(user.username, user.email)){
```

Comment:
do like we do for the arTran and orgSource, your kind of deviating to a special way here. and validation is wrong place to be throwing a DataProblem